esrally/track/loader.py

```diff
@@ -707,8 +725,9 @@ def post_process_for_test_mode(t):
 
 
 class CompleteTrackParams:
-    def __init__(self):
+    def __init__(self, user_specified_track_params=None):
```

Comment:
Should we add:

```python
if user_specified_track_params is None:
    user_specified_track_params = {}
```

This would prevent errors in `unused_user_defined_track_params()` when attempting to access `self.user_specified_track_params.keys()`.

Reply:
Good catch! I addressed in f9621596aff694e5f2d48b70c7d88a1007441b9d and also added a test case.